### PR TITLE
Version update to v2.10.0

### PR DESCRIPTION
**Pull request by mrcnski**

### CODE REVIEW

**Review by interstella-5555 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Changelog
 
+## [2.10.0]
+
+_Note: this version contains breaking changes: the registry entry `revision` is now a `bigint` instead of `number`, and the `executeRequest` method on the client is no longer available._
+
```
Comment (ro-tex) [inline CHANGELOG.md:6]: This means that it shouldn't be version 2.10.0 but 3.0.0.

Comment (mrcnski) [reply CHANGELOG.md:6]: This isn't the first time we had breaking changes in v2 😛

**Review by ro-tex — CHANGES_REQUESTED** [review]
If we're breaking compatibility we need to increment the major version. Unless we've specifically said that we are not following semver.

File: CHANGELOG.md (hunk 1 above)
Comment (ro-tex) [reply CHANGELOG.md:6]: So... no semver? Can we at least announce that somewhere prominent?

I mean, people's `npm` and `yarn` will upgrade this package for them and break their applications, this is not ideal, especially if the fix is a version increment.

**Comment by mrcnski** [comment]
We'll be releasing v3 instead of another minor version. I'll get a couple more breaking changes in before then.

**Pull request closed without merging**